Source/Scene/processPbrMaterials.js

```diff
@@ -697,9 +699,53 @@ define([
             fragmentShader += '    specularIrradiance = mix(specularIrradiance, belowHorizonColor, smoothstep(aroundHorizon, farBelowHorizon, reflectionDotNadir) * inverseRoughness);\n';
             fragmentShader += '    specularIrradiance = mix(specularIrradiance, nadirColor, smoothstep(farBelowHorizon, 1.0, reflectionDotNadir) * inverseRoughness);\n';
 
+            // Luminance model from page 40 of http://silviojemma.com/public/papers/lighting/spherical-harmonic-lighting.pdf
+            fragmentShader += '#ifdef USE_SUN_LUMINANCE \n';
+            // Angle between sun and zenith
+            fragmentShader += '    float LdotZenith = clamp(dot(normalize(czm_inverseViewRotation * l), normalize(v_positionWC * -1.0)), 0.001, 1.0);\n';
+            fragmentShader += '    float S = acos(LdotZenith);\n';
+            // Angle between zenith and current pixel
+            fragmentShader += '    float NdotZenith = clamp(dot(normalize(czm_inverseViewRotation * n), normalize(v_positionWC * -1.0)), 0.001, 1.0);\n';
+            // Angle between sun and current pixel
+            fragmentShader += '    float gamma = acos(NdotL);\n';
+            fragmentShader += '    float numerator = ((0.91 + 10.0 * exp(-3.0 * gamma) + 0.45 * pow(NdotL, 2.0)) * (1.0 - exp(-0.32 / NdotZenith)));\n';
+            fragmentShader += '    float denominator = (0.91 + 10.0 * exp(-3.0 * S) + 0.45 * pow(LdotZenith,2.0)) * (1.0 - exp(-0.32));\n';
+            fragmentShader += '    float luminance = gltf_luminanceAtZenith * (numerator / denominator);\n';
+            fragmentShader += '#endif \n';
+
             fragmentShader += '    vec2 brdfLut = texture2D(czm_brdfLut, vec2(NdotV, 1.0 - roughness)).rg;\n';
             fragmentShader += '    vec3 IBLColor = (diffuseIrradiance * diffuseColor * gltf_iblFactor.x) + (specularIrradiance * SRGBtoLINEAR3(specularColor * brdfLut.x + brdfLut.y) * gltf_iblFactor.y);\n';
-            fragmentShader += '    color += IBLColor;\n';
+
+            fragmentShader += '#ifdef USE_SUN_LUMINANCE \n';
+            fragmentShader += '    color += IBLColor * luminance;\n';
+            fragmentShader += '#else \n';
+            fragmentShader += '    color += IBLColor; \n';
+            fragmentShader += '#endif \n';
+
+            // Environment maps were provided, use them for IBL
+            fragmentShader += '#else \n'; // defined(DIFFUSE_IBL) || defined(SPECULAR_IBL)
+
+            fragmentShader += '    const mat3 yUpToZUp = mat3(-1.0, 0.0, 0.0, 0.0, 0.0, -1.0, 0.0, 1.0, 0.0); \n';
+            fragmentShader += '    vec3 cubeDir = normalize((czm_inverseModelView * vec4(n, 0.0)).xyz); \n';
+            fragmentShader += '    cubeDir = yUpToZUp * cubeDir; \n'; // TODO
```

Comment:
What is this TODO for?

Reply:
I wasn't sure about this. Do all cube maps have the y-axis as the up axis? Should this be configurable?

Reply:
I think so. I checked a couple examples and it seems to be the case.

Reply:
OK. Removed the `TODO`.